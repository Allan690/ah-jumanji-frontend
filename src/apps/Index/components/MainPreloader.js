/* eslint-disable react/no-unescaped-entities */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable react/no-unknown-property */
/* eslint-disable jsx-a11y/label-has-for */
import React, {Component} from 'react';

class MainPreloader extends Component {
  render() {
    return (
      <div id="preloader">
        <div id="section-preloader" className="section-preloader">
          <div className="animation-preloader">
            <div className="txt-loading">
              <span data-text-preloader="A" className="letters-loading">A</span>
              <span data-text-preloader="u" className="letters-loading">u</span>
              <span data-text-preloader="t" className="letters-loading">t</span>
              <span data-text-preloader="h" className="letters-loading">h</span>
              <span data-text-preloader="o" className="letters-loading">o</span>
              <span data-text-preloader="r" className="letters-loading">r</span>
              <span data-text-preloader="s" className="letters-loading">s</span>
              <span data-text-preloader=" " className="letters-loading" />
              <span data-text-preloader="H" className="letters-loading">H</span>
              <span data-text-preloader="a" className="letters-loading">a</span>
              <span data-text-preloader="v" className="letters-loading">v</span>
              <span data-text-preloader="e" className="letters-loading">e</span>
              <span data-text-preloader="n" className="letters-loading">n</span>
            </div>
          </div>
          <div className="loader-section" />
        </div>
      </div>
    );
  }
}

export default MainPreloader;


