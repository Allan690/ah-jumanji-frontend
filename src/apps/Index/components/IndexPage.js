import React, {Component} from 'react';
import LoginContainer from '../../Login/components/LoginContainer';
import MainPreloader from './MainPreloader';


class IndexPage extends Component {
  constructor(props) {
    super(props);
    this.state = {isLoading: true};
  }

  componentDidMount() {
    // the setTimeout just simulates an async action, after which the component will render the content
    setTimeout(() => this.setState({isLoading: false}), 1000);
  }

  render() {
    let {isLoading} = this.state;
    return (
      isLoading ? (
        <MainPreloader />) : (
          <div>
            <LoginContainer />
          </div>
      )
    );
  }
}


export default IndexPage;
