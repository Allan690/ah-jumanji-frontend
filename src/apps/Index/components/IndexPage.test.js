import React from 'react';

import { shallow } from 'enzyme';

import IndexPage from './IndexPage';

import MainPreloader from './MainPreloader';

describe('IndexPage', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<IndexPage debug />);

    expect(component).toMatchSnapshot();
  });
});

describe('MainPreloader', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<MainPreloader debug />);

    expect(component).toMatchSnapshot();
  });
});
