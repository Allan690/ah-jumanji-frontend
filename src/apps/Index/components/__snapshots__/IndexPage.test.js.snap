// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage should render correctly in "debug" mode 1`] = `<MainPreloader />`;

exports[`MainPreloader should render correctly in "debug" mode 1`] = `
<div
  id="preloader"
>
  <div
    className="section-preloader"
    id="section-preloader"
  >
    <div
      className="animation-preloader"
    >
      <div
        className="txt-loading"
      >
        <span
          className="letters-loading"
          data-text-preloader="A"
        >
          A
        </span>
        <span
          className="letters-loading"
          data-text-preloader="u"
        >
          u
        </span>
        <span
          className="letters-loading"
          data-text-preloader="t"
        >
          t
        </span>
        <span
          className="letters-loading"
          data-text-preloader="h"
        >
          h
        </span>
        <span
          className="letters-loading"
          data-text-preloader="o"
        >
          o
        </span>
        <span
          className="letters-loading"
          data-text-preloader="r"
        >
          r
        </span>
        <span
          className="letters-loading"
          data-text-preloader="'"
        >
          '
        </span>
        <span
          className="letters-loading"
          data-text-preloader="s"
        >
          s
        </span>
        <span
          className="letters-loading"
          data-text-preloader=" "
        >
           
        </span>
        <span
          className="letters-loading"
          data-text-preloader="H"
        >
          H
        </span>
        <span
          className="letters-loading"
          data-text-preloader="a"
        >
          a
        </span>
        <span
          className="letters-loading"
          data-text-preloader="v"
        >
          v
        </span>
        <span
          className="letters-loading"
          data-text-preloader="e"
        >
          e
        </span>
        <span
          className="letters-loading"
          data-text-preloader="n"
        >
          n
        </span>
      </div>
    </div>
    <div
      className="loader-section"
    />
  </div>
</div>
`;
